ncc: add page (#5447)
tldr-pages/tldr: pages/common/ncc.md

# ncc

> > Compile a Node.js project into a single file.

Reviewer: ```suggestion
> Compile a Node.js application into a single file.
```

> > Supports TypeScript, binary addons and dynamic requires.

Reviewer 2: `dynamic requires` sounds a litttle strange. Should this mean `dynamic requirements` or is this correct?

Reviewer: In Node.js (only, not browser JavaScript :/), there's a function called `require()`, and it's used to import classes, functions, variables, and any other value from other `.js` files (exported via the JavaScript object `module.exports`).  A sample would look like this:

- `hello.js`

  ```js
  function hello() {
    return "Hello, World!";
  }
  module.exports = hello;
  ```

- `example.js`

  ```js
  const hello = require("./hello"); // in Node.js, the .js extension is implied and usually omitted
  console.log(hello());
  ```

- Running:

  ```console
  $ node example.js
  Hello, World!
  ```

"dynamic require" here is referring to the fact that, since `require` is just a normal function, it can be used dynamically. Use cases include conditionally `require()`ing a module, looping over a list of files and `require()`ing them (e.g. to load plugins, or load configuration JSON files (because `require()` supports `.json` files too)).  You can also pass variables to it, not just hardcoded strings.

Reviewer: We could possibly use `... dynamic require()s.`? Although I don't love that.

Reviewer 2: Thanks for the explaination. I'd say it's fine like this.

> More information: <https://github.com/vercel/ncc>.

> - Bundle a JavaScript application:

Reviewer: ```suggestion
- Bundle a Node.js application:
```

`ncc build {{path/to/file.js}}`

> - Bundle and minify a JavaScript application:

Reviewer: ```suggestion
- Bundle and minify a Node.js application:
```

`ncc build --minify {{path/to/file.js}}`

> - Bundle and minify a JavaScript application with source maps:

Reviewer: ```suggestion
- Bundle and minify a Node.js application and generate source maps:
```

`ncc build --source-map {{path/to/file.js}}`

> - Automatically recompile on changes to application files:

Reviewer: ```suggestion
- Automatically recompile on changes to source files:
```

`ncc build --watch {{path/to/file.js}}`

> - Bundle into a temporary directory and run a JavaScript application for testing:

Reviewer: ```suggestion
- Bundle a Node.js application into a temporary directory and run it for testing:
```
